kie-pmml-trusty/kie-pmml-compiler/kie-pmml-compiler-commons/src/main/java/org/kie/pmml/compiler/commons/utils/ModelUtils.java

```diff
@@ -343,6 +375,105 @@ public static DATA_TYPE getDataType(DataDictionary dataDictionary, String target
+        String name = toConvert.getName() != null ? toConvert.getName().getValue() : "" + toConvert.hashCode();
+        KiePMMLOutputField.Builder builder = KiePMMLOutputField.builder(name, Collections.emptyList());
+        final String targetField = toConvert.getTargetField() != null ? toConvert.getTargetField().getValue() : null;
+        if (targetField != null) {
+            builder.withTargetField(targetField);
+        }
+        final RESULT_FEATURE resultFeature = toConvert.getResultFeature() != null ?
+                RESULT_FEATURE.byName(toConvert.getResultFeature().value()) : null;
+        if (resultFeature != null) {
+            builder = builder.withResultFeature(resultFeature);
+        }
+        if (toConvert.getRank() != null) {
+            builder = builder.withRank(toConvert.getRank());
+        }
+        if (toConvert.getExpression() != null) {
+            KiePMMLExpression kiePMMLExpression = convertToKiePMMLExpression(toConvert.getExpression());
+            builder = builder.withKiePMMLExpression(kiePMMLExpression);
+        }
+        return builder.build();
+    }
+
+    public static KiePMMLExpression convertToKiePMMLExpression(Expression toConvert) {
+        KiePMMLExpression toReturn = null;
+        String expressionType = toConvert.getClass().getSimpleName();
+        switch (expressionType) {
+            case "Apply":
+                toReturn = convertToKiePMMLApply((Apply) toConvert);
+                break;
+            case "Constant":
+                toReturn = convertToKiePMMLConstant((Constant) toConvert);
+                break;
+            case "FieldRef":
+                toReturn = convertToKiePMMLFieldRef((FieldRef) toConvert);
+                break;
+            default:
+                // Not implemented, yet
+
+        }
+        return toReturn;
+    }
+
+    public static KiePMMLApply convertToKiePMMLApply(Apply toConvert) {
+        String name = "" + toConvert.hashCode();
+        KiePMMLApply.Builder builder = KiePMMLApply.builder(name, Collections.emptyList(), toConvert.getFunction());
+        if (toConvert.getExpressions() != null) {
+            List<KiePMMLExpression> kiePMMLExpressions = toConvert.getExpressions().stream()
+                    .map(ModelUtils::convertToKiePMMLExpression)
+                    .collect(Collectors.toList());
+            builder = builder.withKiePMMLExpressions(kiePMMLExpressions);
+        }
+        if (toConvert.getMapMissingTo() != null) {
+            builder = builder.withMapMissingTo(toConvert.getMapMissingTo());
+        }
+        if (toConvert.getDefaultValue() != null) {
+            builder = builder.withDefaultValue(toConvert.getDefaultValue());
+        }
+        if (toConvert.getInvalidValueTreatment() != null) {
+            builder =
+                    builder.withInvalidValueTreatmentMethod(INVALID_VALUE_TREATMENT_METHOD.byName(toConvert.getInvalidValueTreatment().value()));
+        }
```

Comment:
What about move all `if` condition inside the `with*` method?
In this way you can have something similar to
```java
builder.withKiePMMLExpressions(kiePMMLExpressions)
   .withMapMissingTo(toConvert.getMapMissingTo())
   .withDefaultValue(toConvert.getDefaultValue())
withInvalidValueTreatmentMethod(INVALID_VALUE_TREATMENT_METHOD.byName(toConvert.getInvalidValueTreatment().value()))
``

Reply:
@danielezonca 
done (can not completely delegate everything to builder because I do not want to add explicit dependency on external library)